File: admin_app/admin_flutter/lib/models/master_email_setting.dart
class MasterEmailSetting {
  const MasterEmailSetting({
    required this.authMethod,
    this.smtpHost,
    required this.smtpPort,
    this.username,
    this.clientId,
    this.fromEmail,
    this.fromName,
    required this.enabled,
    required this.passwordConfigured,
    required this.clientSecretConfigured,
    required this.refreshTokenConfigured,
    required this.configured,
  });

  final String authMethod;
  final String? smtpHost;
  final int smtpPort;
  final String? username;
  final String? clientId;
  final String? fromEmail;
  final String? fromName;
  final bool enabled;
  final bool passwordConfigured;
  final bool clientSecretConfigured;
  final bool refreshTokenConfigured;
  final bool configured;

  factory MasterEmailSetting.fromJson(Map<String, dynamic> json) {
    return MasterEmailSetting(
      authMethod: json['auth_method'] as String? ?? 'smtp',
      smtpHost: json['smtp_host'] as String?,
      smtpPort: json['smtp_port'] as int? ?? 587,
      username: json['username'] as String?,
      clientId: json['client_id'] as String?,
      fromEmail: json['from_email'] as String?,
      fromName: json['from_name'] as String?,
      enabled: json['enabled'] == true,
      passwordConfigured: json['password_configured'] == true,
      clientSecretConfigured: json['client_secret_configured'] == true,
      refreshTokenConfigured: json['refresh_token_configured'] == true,
      configured: json['configured'] == true,
    );
  }
}

class MasterEmailSettingInput {
  const MasterEmailSettingInput({
    required this.authMethod,
    required this.smtpHost,
    required this.smtpPort,
    required this.username,
    this.password,
    this.clientId,
    this.clientSecret,
    this.refreshToken,
    required this.fromEmail,
    required this.fromName,
    required this.enabled,
  });

  final String authMethod;
  final String smtpHost;
  final int smtpPort;
  final String username;
  final String? password;
  final String? clientId;
  final String? clientSecret;
  final String? refreshToken;
  final String fromEmail;
  final String fromName;
  final bool enabled;

  Map<String, dynamic> toJson() => {
    'auth_method': authMethod,
    'smtp_host': smtpHost,
    'smtp_port': smtpPort,
    'username': username,
    'password': password,
    'client_id': clientId,
    'client_secret': clientSecret,
    'refresh_token': refreshToken,
    'from_email': fromEmail,
    'from_name': fromName,
    'enabled': enabled,
  };
}

File: backend/app/models/master_email_setting.py
from datetime import datetime

from sqlalchemy import Boolean, DateTime, Integer, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column

from app.core.master_database import MasterBase


class MasterEmailSetting(MasterBase):
    __tablename__ = "master_email_settings"

    provider: Mapped[str] = mapped_column(String(40), primary_key=True)
    auth_method: Mapped[str] = mapped_column(String(30), nullable=False, default="smtp")
    smtp_host: Mapped[str | None] = mapped_column(String(180), nullable=True)
    smtp_port: Mapped[int] = mapped_column(Integer, nullable=False, default=587)
    username: Mapped[str | None] = mapped_column(String(180), nullable=True)
    password_encrypted: Mapped[str | None] = mapped_column(Text, nullable=True)
    client_id: Mapped[str | None] = mapped_column(String(240), nullable=True)
    client_secret_encrypted: Mapped[str | None] = mapped_column(Text, nullable=True)
    refresh_token_encrypted: Mapped[str | None] = mapped_column(Text, nullable=True)
    from_email: Mapped[str | None] = mapped_column(String(180), nullable=True)
    from_name: Mapped[str | None] = mapped_column(String(120), nullable=True)
    enabled: Mapped[bool] = mapped_column(Boolean, nullable=False, default=False)
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), onupdate=func.now()
    )
